WMF Framework/Significant Events Endpoint/ArticleAsLivingDocViewModels.swift

```diff
@@ -844,10 +854,96 @@ public extension ArticleAsLivingDocViewModel.Event.Large {
+        let referenceTitleAttributes = [NSAttributedString.Key.font: referenceTitleFont]
+        
+        let threeLineSnippetText = """
+                                1
+                                2
+                                3
+                            """
+        let oneLineSnippetText = "1"
+        
+        let threeLineAttString = NSAttributedString(string: threeLineSnippetText, attributes: snippetAttributes)
+        let oneLineAttString = NSAttributedString(string: oneLineSnippetText, attributes: referenceTitleAttributes)
+        
+        let threeLineSnippetHeight = ceil(threeLineAttString.boundingRect(with: CGSize(width: Self.availableSideScrollingCellWidth, height: CGFloat.infinity), options: [.usesLineFragmentOrigin], context: nil).height)
+        let oneLineSnippetHeight = ceil(oneLineAttString.boundingRect(with: CGSize(width: Self.availableSideScrollingCellWidth, height: CGFloat.infinity), options: [.usesLineFragmentOrigin], context: nil).height)
+        
+        var tallestSnippetChangeDetailHeight: CGFloat = 0
+        var tallestReferenceChangeDetailHeight: CGFloat = 0
+        let availableWidth = Self.availableSideScrollingCellWidth - Self.padding.left - Self.padding.right
+        changeDetails.forEach { (changeDetail) in
+            switch changeDetail {
+            case .snippet(let snippet):
+                let heightOfEntireSnippet = ceil(snippet.displayText.boundingRect(with: CGSize(width: availableWidth, height: CGFloat.infinity), options: [.usesLineFragmentOrigin], context: nil).height) + Self.padding.top + Self.padding.bottom
+                
+                if tallestSnippetChangeDetailHeight < heightOfEntireSnippet {
+                    tallestSnippetChangeDetailHeight = heightOfEntireSnippet
+                }
+                
+            case .reference(let reference):
+                let titleHeight = oneLineSnippetHeight
+                let descriptionHeight = ceil(reference.description.boundingRect(with: CGSize(width: availableWidth, height: CGFloat.infinity), options: [.usesLineFragmentOrigin], context: nil).height)
+                let totalHeight = Self.padding.top + titleHeight + Self.changeDetailReferenceTitleDescriptionSpacing + descriptionHeight + Self.padding.bottom
+                
+                if tallestReferenceChangeDetailHeight < totalHeight {
+                    tallestReferenceChangeDetailHeight = totalHeight
+                }
+            }
+        }
+
+        var nearlyFinalHeight: CGFloat = 0
+        if tallestSnippetChangeDetailHeight == 0 {
+            nearlyFinalHeight = tallestReferenceChangeDetailHeight
+        } else if tallestReferenceChangeDetailHeight == 0 {
+            nearlyFinalHeight = min(tallestSnippetChangeDetailHeight, threeLineSnippetHeight + Self.padding.top + Self.padding.bottom)
+        } else {
+            let finalSnippetHeight = min(tallestSnippetChangeDetailHeight, threeLineSnippetHeight)
```

Comment:
do we need to add top/bottom padding here, like 2 lines above?

Reply:
Yeah possibly, I'll take a look when I do my sizing pass.